packages/codemirror-extension/src/keymaps.ts

```diff
@@ -0,0 +1,265 @@
+        let currentCell = notebookTracker.activeCell;
+        // TODO: these references will be undefined
+        // Depending what our last motion was, we may want to do different
+        // things. If our last motion was moving vertically, we want to
+        // preserve the HPos from our last horizontal move.  If our last motion
+        // was going to the end of a line, moving vertically we should go to
+        // the end of the line, etc.
+        switch (vim.lastMotion) {
+          case 'moveByLines':
+          case 'moveByDisplayLines':
+          case 'moveByScroll':
+          case 'moveToColumn':
+          case 'moveToEol':
+          // JUPYTER PATCH: add our custom method to the motion cases
+          // eslint-disable-next-line no-fallthrough
+          case 'moveByLinesOrCell':
+            endCh = vim.lastHPos;
+            break;
+          default:
+            vim.lastHPos = endCh;
+        }
+        let repeat = motionArgs.repeat + (motionArgs.repeatOffset || 0);
+        let line = motionArgs.forward ? cur.line + repeat : cur.line - repeat;
+        let first = cm.firstLine();
+        let last = cm.lastLine();
+        // Vim cancels linewise motions that start on an edge and move beyond
+        // that edge. It does not cancel motions that do not start on an edge.
+
+        // JUPYTER PATCH BEGIN
+        // here we insert the jumps to the next cells
+        if (line < first || line > last) {
+          // var currentCell = ns.notebook.get_selected_cell();
+          // var currentCell = tracker.activeCell;
+          // var key = '';
+          if (currentCell?.model.type === 'markdown') {
+            (currentCell as MarkdownCell).rendered = true;
+          }
+          if (motionArgs.forward) {
+            void commands.execute('notebook:move-cursor-down');
+          } else {
+            void commands.execute('notebook:move-cursor-up');
+          }
+          return;
+        }
+        vim.lastHSPos = cm.charCoords(CodeMirror.Pos(line, endCh), 'div').left;
+        return (CodeMirror as any).Pos(line, endCh);
+      }
+    );
+    vim.mapCommand(
+      'k',
+      'motion',
+      'moveByLinesOrCell',
+      { forward: false, linewise: true },
+      { context: 'normal' }
+    );
+    vim.mapCommand(
+      'j',
+      'motion',
+      'moveByLinesOrCell',
+      { forward: true, linewise: true },
+      { context: 'normal' }
+    );
+    CodeMirror.prototype.save = () => {
+      void commands.execute('docmanager:save');
+    };
+    vim.defineEx('quit', 'q', function (cm: any) {
+      void commands.execute('notebook:enter-command-mode');
+    });
+    vim.defineAction('splitCell', (cm: any, actionArgs: any) => {
+      void commands.execute('notebook:split-cell-at-cursor');
+    });
+    vim.mapCommand('-', 'action', 'splitCell', {}, { extra: 'normal' });
+    keymapRemoval = [
+      commands.addKeyBinding({
+        selector: '.jp-Notebook.jp-mod-editMode',
+        keys: ['Ctrl J'],
```

Comment:
Is `Ctrl` also the expected key to use here for Mac people? Or would they use `Cmd` for this, so `Accel` might be better? I don't use Mac much so I don't know myself.

Reply:
Don't have a mac so I never thought about this. This is what the `jupyterlab_vim` extension currently does so I just continued that. https://github.com/axelfahy/jupyterlab-vim/blob/c4e43f940ef4be4c961608b6192412d2f3a33d1f/src/index.ts#L466-L470

probably could be improved though

Reply:
:+1:  let's keep the same binding as what the `jupyter_lab_vim` extension had since that's what people would be used to.